src/client/Microsoft.Identity.Client.Broker/RuntimeBroker.cs

```diff
@@ -77,25 +78,38 @@ internal class RuntimeBroker : IBroker
             {
                 //Login Hint
                 string loginHint = authenticationRequestParameters.LoginHint ?? authenticationRequestParameters?.Account?.Username;
-
                 _logger.Verbose("[WamBroker] AcquireTokenInteractive - login hint provided? " + string.IsNullOrEmpty(loginHint));
 
-                using (var result = await core.SignInInteractivelyAsync(
-                    _parentHandle,
-                    authParams,
-                    authenticationRequestParameters.CorrelationId.ToString("D"),
-                    loginHint,
-                    cancellationToken).ConfigureAwait(false))
+                //if login hint is not provided or if account is null
+                if (string.IsNullOrEmpty(authenticationRequestParameters.LoginHint))
                 {
-                    if (result.IsSuccess)
+                    using (var result = await core.SignInInteractivelyAsync(
+                        _parentHandle,
+                        authParams,
+                        authenticationRequestParameters.CorrelationId.ToString("D"),
+                        loginHint,
+                        cancellationToken).ConfigureAwait(false))
                     {
-                        msalTokenResponse = WamAdapters.ParseRuntimeResponse(result, authenticationRequestParameters, _logger);
-                        _logger.Verbose("[WamBroker] Successfully retrieved token.");
+                        msalTokenResponse = WamAdapters.AcquireTokenFromBroker(result, authenticationRequestParameters, _logger, errorMessage);
+
                     }
-                    else
+                }
+                //if an account is already signed in
+                else
+                {
+                    using (var wamAccount = await core.ReadAccountByIdAsync(
+                    authenticationRequestParameters.Account.HomeAccountId.ObjectId,
```

Comment:
I think this logic needs to be cleaned up a bit. It should more like: 

1. If an Account object is present and if that object has an ID and Object Id THEN do `AcquireTokenInteractivelyAsync`
2. If a loginHint is present (as defined at line 80), then do SignInInteractivelyAsync

Remember that we don't force app developers to use our Account objects. THey only have an interface called IAccount (in hinghsight, this was a mistake). I have seen a LOT of ppl creating their own Account objects which ignore the ID.